### DISCUSSION TITLE
Sid Detection in Mod Player

### DISCUSSION THREAD

**Discussion by jajpohke** [frntc/Sidekick64#80] [General]
I've experienced on a Breadbin of mine (Assy 250425)  that SID is not correctly detected as his 6581 (original)  but the player icons shows "dual sid" so I miss come channels in reproducion when I play some mod in Music Player.

This specific board has been repaired  replacing PLA with a modern replacement listed on ebay and I had also to replace both 4066s cmos quat biswitch.

The same Sidekick64 Board (BOM 0.42+Rpi3A+) has been pushed in a C64C shortboard where 8580 is correcly as in another Breadbin (assy 250407) where the 6581 original too is detected.

I dunno if its that breadbin ussue or common, but is there a way to force it and specify which Sid is installed? (RAD Doom also give unknown sid in options with the first scenario, havent tested it on other machines).

Thanx

**Comment by frntc** [comment]
The MOD-player (and also RAD-Doom) use this method for detecting the SID-type: https://codebase64.org/doku.php?id=base:detecting_sid_type_-_safe_method

Is that PLA doing something special? If two SIDs are detected, then maybe the SID reacts to different addresses? Maybe you can swap PLA/SID with your other longboard to figure out where the problem ist.

Another experiment would be to use some other program which automatically detects the SID (the MOD-player code is running on the 6510, so it's pretty independent of Sidekick64)

Enforcing SID-type: I haven't thought about this (wanted to keep it as simple as possible), and that SID-detect is actually called the "safe method" :-) let's first see where exactly the problem is!

**Comment by jajpohke** [comment]
I did some homeworks as you suggested, because you are the Master,  during my launchbreak and the winner is.....the PLA from eBay! But there something I dont understand and cealing behind the door or inside this addressing IC....  

If I put the ebay's PLA on 250407 longboard SID detection is ok, I've all channels and its pretty good; also,  if I put 250407's PLA on 250425 longboard SID detection is ok too so I assume that cant be a different PLA/compatibilty scenario; I did swap back in order to assume that was a contact/socket issue but no, the starting issue scenario with double sid detected came back, so I did a permanet swap).

I also did SID swap but nothing changes; issue has been solved and, somehow, described on github if someone could experiment this in future, so I'm pretty happy atm (even if I pray to dont burn the original PLA now installed on 250425 with a big heatsink).

Anyway, tbh,  I dont think that add an option to disable SID detection and do "run as is" was a bad and complex idea for next built ;-)

btw, you are a very kind person as all comunities reports, so thank you again for quick support (even if I'm bit a lurker ;-) ); let me know if I can do something more before put these longboard machines in their boxes, and turn page.

**Comment by frntc** [comment]
On these longboards the SID-chip select signal comes from a 74LS139/U15 which demultiplexes with signals from the PLA. Some replacement PLAs have slightly off timings or voltage levels, so this interplay might cause problems on one board and not on the other. (you could also swap the LS139, but it's not worth the time -- I'd say case closed ;-))

Regarding options: any good ideas how to select options and communicate them are welcome (you notices that there's already an alternative visualization option on PAL machines?)
